### PR TITLE
huge workflow re-arrange

### PR DESCRIPTION
**Pull request by MandyMeindersma**
Okay so my earlier PR fixed the workflow, but like not well.

This is a wayyyy better way for everything to run. Also now we actually scrape the research and google pages in the workflow. We also actually post the faculty page now (before we forgot)

Please go over this with a fin tooth comb. I changed a lot and it runs correctly but there is a chance I could have pooched something

### CODE REVIEW

File: web/bfex/components/data_pipeline/tasks/googlescholar.py [hunk 1]
```diff
@@ -34,31 +28,31 @@ def run(self, data):
...
+        search_dup = Document.search().query('match', faculty_id=faculty.faculty_id).query("match", source="GoogleScholar")
+        search_dup.delete()
+
+        faculty = search_results[0]
+        if faculty.google_scholar is not None and "http" in faculty.google_scholar:
+            scraper = ScraperFactory.create_scraper(faculty.google_scholar, ScraperType.GOOGLESCHOLAR)
+            scrapps = scraper.get_scrapps()
+            for scrapp in scrapps:
+                doc = Document()
+                doc.source = "GoogleScholar"
```
Comment (GarryBullock) [inline web/bfex/components/data_pipeline/tasks/googlescholar.py]: I thought this was going to be a tuple with a string value? :P

File: web/bfex/components/data_pipeline/tasks/persist.py [hunk 2]
```diff
@@ -70,6 +70,7 @@ def run(self, data):
             doc.save()
 
         faculty.save()
+        print("persist")
```
Comment (GarryBullock) [inline web/bfex/components/data_pipeline/tasks/persist.py]: Let's remove any print statements.

File: web/bfex/components/data_pipeline/tasks/researchid.py [hunk 3]
```diff
@@ -34,60 +28,54 @@ def run(self, data):
...
-                
-                scraper = ScraperFactory.create_scraper(faculty.research_id, ScraperType.RESEARCHID)
-                scrapps = scraper.get_scrapps()
+        search_results = Faculty.search().query('match', name=faculty_name).execute()
+        if len(search_results) > 1:
+            # Shouldn't happen, but could.
+            raise WorkflowException("Professor id is ambiguous during search ... More than 1 result")
+
+        search_dup = Document.search().query('match', faculty_id=faculty.faculty_id).query("match", source="ResearchId")
+        search_dup.delete()
```
Comment (GarryBullock) [inline web/bfex/components/data_pipeline/tasks/researchid.py]: Are you using this later? Does it need to be stored? You can just put Document...delete()

File: web/bfex/components/data_pipeline/tasks/researchid.py [hunk 4]
```diff
@@ -34,60 +28,54 @@ def run(self, data):
...
+            doc.source = "ResearchId"
+            try:
+                doc.text = keywords_and_description.meta_data["description"]
+            except:
+                print("No description")
+                doc.text = ""
+            try:
+                doc.user_keywords = keywords_and_description.meta_data["keywords"]
+            except:
+                print("No keywords")
```
Comment (GarryBullock) [inline web/bfex/components/data_pipeline/tasks/researchid.py]: Should this also get a default value?

File: web/bfex/components/data_pipeline/tasks/scrape.py [hunk 5]
```diff
@@ -32,16 +33,39 @@ def run(self, data):
...
+        search_dup = Document.search().query('match', faculty_id=faculty.faculty_id).query("match", source="FacultyPage")
+        search_dup.delete()
+            
+        faculty_directory_url = URLs.build_faculty_url(faculty_name)
 
         scraper = ScraperFactory.create_scraper(faculty_directory_url, ScraperType.PROFILE)
         scrapp = scraper.get_scrapps()[0]
 
+        doc = Document()
+        doc.source = "FacultyPage"
```
Comment (GarryBullock) [inline web/bfex/components/data_pipeline/tasks/scrape.py]: Weren't we calling this profile before? Has it been adjusted wherever else it was being used?

File: web/bfex/components/data_pipeline/tasks/researchid.py (hunk 3 above)
Comment (MandyMeindersma) [reply web/bfex/components/data_pipeline/tasks/researchid.py]: pep8 line too long?

File: web/bfex/components/data_pipeline/tasks/scrape.py (hunk 5 above)
Comment (MandyMeindersma) [reply web/bfex/components/data_pipeline/tasks/scrape.py]: I thought @morlick was calling profile the thing from research id, let me invesgigate

File: web/bfex/components/data_pipeline/tasks/persist.py (hunk 2 above)
Comment (MandyMeindersma) [reply web/bfex/components/data_pipeline/tasks/persist.py]: kek alreadt have a commit with that

File: web/bfex/components/data_pipeline/tasks/googlescholar.py (hunk 1 above)
Comment (MandyMeindersma) [reply web/bfex/components/data_pipeline/tasks/googlescholar.py]: yeah we messed the doc up

File: web/bfex/components/data_pipeline/tasks/researchid.py (hunk 3 above)
Comment (GarryBullock) [reply web/bfex/components/data_pipeline/tasks/researchid.py]: You can break up lines on the . so:

Document.search()
                 .query(...)
                 .query()

etc.

Comment (MandyMeindersma) [reply web/bfex/components/data_pipeline/tasks/researchid.py]: kek

File: web/bfex/components/data_pipeline/tasks/scrape.py (hunk 5 above)
Comment (MandyMeindersma) [reply web/bfex/components/data_pipeline/tasks/scrape.py]: ugh, no garry I am dumb

Comment (MandyMeindersma) [reply web/bfex/components/data_pipeline/tasks/scrape.py]: omg you already coded these 4 lines, deleted now. I am so dumb

File: web/bfex/components/data_pipeline/tasks/researchid.py (hunk 4 above)
Comment (MandyMeindersma) [reply web/bfex/components/data_pipeline/tasks/researchid.py]: meh, if its empty then no keys words, who cares?

Comment (GarryBullock) [reply web/bfex/components/data_pipeline/tasks/researchid.py]: What happens if you try and access it after and it doesn't have any. Is it just an empty string? I just don't want a surprise exception later on.

Comment (MandyMeindersma) [reply web/bfex/components/data_pipeline/tasks/researchid.py]: yeah, ill add more

**Comment by GarryBullock** [comment]
LGTM, will merge once our build is fixed.

**Pull request merged**